modules/align/main.nf

```diff
@@ -0,0 +1,75 @@
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process ALIGN {
+    tag "$meta"
+    label 'process_low'
+    publishDir "${params.out}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename,
+                                        options:params.options, 
+                                        publish_dir:"${task.process}".replace(':','/').toLowerCase(), 
+                                        publish_id:meta) }
+    input:
+        tuple val(meta), path(reads), path(reference)
+
+    output:
+        tuple val(meta), path("*.sam"), emit: alignment
+
+    script:
+        """
+        minimap2 -t $task.cpus \\
```

Comment:
You need to use the minimap modules here for this, and then you can combine this into a subworkflow if you want.

Reply:
Ok, I will do that.